packages/@react-spectrum/avatar/src/Avatar.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {classNames, dimensionValue, useDOMRef, useStyleProps} from '@react-spectrum/utils';
+import {DOMRef} from '@react-types/shared';
+import React, {forwardRef} from 'react';
+import {SpectrumAvatarProps} from '@react-types/avatar';
+import styles from '@adobe/spectrum-css-temp/components/avatar/vars.css';
+import {useAvatar} from '@react-aria/avatar';
+import {useProviderProps} from '@react-spectrum/provider';
+
+const DEFAULT_SIZE = 'avatar-size-100';
+const SIZE_RE = /^size-\d+/;
+
+function Avatar(props: SpectrumAvatarProps, ref: DOMRef<HTMLImageElement>) {
+  const {
+    isDisabled,
+    size,
+    src,
+    ...otherProps
+  } = useProviderProps(props);
+
+  const {styleProps} = useStyleProps(otherProps);
+  const domRef = useDOMRef(ref);
+  const {avatarProps} = useAvatar(props);
+  const {alt, ...otherAvatarProps} = avatarProps; // alt for jsx-a11y/alt-text
+
+  const sizeValue = typeof size !== 'number' && (SIZE_RE.test(size) || !isNaN(size as any))
```

Comment:
doesn't `typeof size !== 'number'` prevent `!isNaN(size)` from ever being true? or am i confused with double negatives?

Reply:
I agree that this part is not super clear. Added a comment.
Basically, we want to cover the case where `size` is a `string` number, such as `"300"` (as opposed to a `number` number; e.g. `300`). In this case, `size` is not of type `number`, but `isNaN` should still return `false`:
```ts
isNaN(300); // false, because 300 is a number
isNaN('300'); // false, because '300' is, essentially, a number
isNaN('300px'); // true, because '300px' is not a number
```
Also, casting `size` as `any` here is a hack since `isNaN` expects `number`, but I want it to handle the `string` number.

Reply:
Just wanna make sure: should passing `size="300"` work? Doesn't seem like it does locally